module-check/src/main/clojure/clojure/core/typed/base_env.clj

```diff
@@ -1443,7 +1443,17 @@ clojure.core/rand-nth (All [x] [(U (Indexed x) (SequentialSeqable x)) -> x])
   this-ns
 clojure.set/union (All [x] [(Set x) * -> (Set x)])
 clojure.set/intersection (All [x] [(Set x) (Set x) * -> (Set x)])
+
 clojure.set/difference (All [x] [(Set x) (Set Any) * -> (Set x)])
+clojure.set/subset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/superset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/join [(Set (Map Any Any)) (Set (Map Any Any)) -> (Set (Map Any Any))]
+clojure.set/index [(Set (Map Any Any)) (Vec Any) -> (Map (Map Any Any) (Set (Map Any Any)))]
+clojure.set/map-invert (All [a b] [(Map a b) -> (Map b a)])
+clojure.set/project [(Set (Map Any Any)) (Vec Any) -> (Set (Map Any Any))]
+clojure.set/rename [(Set Any) (Map Any Any) -> (Set Any)]
+clojure.set/rename-keys [(Map Any Any) (Map Any Any) -> (Map Any Any)]
+clojure.set/select [ (IFn [AnyInteger -> java.lang.Boolean]) (Set Any) -> (Set Any)]
```

Comment:
`select` takes 2 argument - function and collection of elements. The first argument, function, takes an element and return boolean. And element types is not restricted to integer. It can be anything. And second argument is not necessary Set, it can be any collection, so we should use `Seq` instead. 

I'd suggest using something like
```clojure
(All [a] [(IFn [a -> Boolean]) (Seq a) -> (Set a)])
```